tensorflow/inference/docker/build_artifacts/sagemaker/python_service.py

```diff
@@ -493,15 +499,14 @@ def validate_model_versions(self, versions):
 
     def _upload_mme_instance_status(self):
         log.info("uploaded mme instance status file with content: {}".format(self._mme_tfs_instances_status))
-        with open(MME_TFS_INSTANCE_STATUS_FILE, 'wb') as handle:
-            pickle.dump(self._mme_tfs_instances_status, handle, protocol=pickle.HIGHEST_PROTOCOL)
+        shared_tfs_instance_dict.clear()
+        for key, value in self._mme_tfs_instances_status.items():
+            shared_tfs_instance_dict[key] = pickle.dumps(value)
```

Comment:
I'm not convinced that we need the pickling here at all. I think we could instead just put `value` directly in the manager dict and it would work.